### DISCUSSION TITLE
No trending posts?

### DISCUSSION THREAD

**Discussion by limdingwen** [mastodon/mastodon#21208] [Q&A]
I’m on a very small server, with a relay. Seems like there’s no trending posts, and the language selector has nothing inside it as well. Is this intended?

**Comment by limdingwen** [comment]
![CFD11CEF-490E-448E-8368-9455D493C9EB](https://user-images.githubusercontent.com/1744967/202899362-dea65fcc-5fd8-41b6-af9f-c3ec6b9f7aa7.png)

Sorry, not too sure how to edit the post. Image uploading failed the first time.

**Reply by loremipsum2022** [reply]
Did you solve that issue? same here :/

**Reply by melroyvandenberg** [reply]
Maybe configuring a relay might help??

**Comment by themullinator** [comment]
For instances with at least 5 members, it's unlikely you'll get any trending posts unless 5+ people all reblog the same post within a couple hours of each other. 

For instances with fewer than 5 members, you will never get any trending posts.

This is due to how trending is implemented: https://github.com/mastodon/mastodon/blob/v4.0.2/app/models/trends/statuses.rb#L7 sets a threshold of 5 reblogs or it defaults to a trend score of zero: https://github.com/mastodon/mastodon/blob/v4.0.2/app/models/trends/statuses.rb#L103 (this helps preserve privacy on micro-instances where it would otherwise be really easy to have every single thing that 2-3 people reblog end up as trend.)

Because there are currently no posts above-threshold on your instance, there are no unapproved trending posts to filter by language, which is why that dropdown is empty. That feels like a somewhat unintuitive experience though, and may be worth a bug report.

**Reply by melroyvandenberg** [reply]
What about trending posts from other M. servers? Do I need a relay to get trending posts then?

**Reply by SiskoUrso** [reply]
This does not appear to be the case, have my server which is less than 5 members on multiple relays with many servers and the top posts have never shown anything after a month of running the instance.

**Reply by melroyvandenberg** [reply]
ps. How did you get multiple relays working? I added also several relays.. but Only 1 accepted the request. The rests are still all 'pending' too bad.. :(

**Reply by SiskoUrso** [reply]
When you are adding relays make sure you are following the guidelines they may have , some of them require your server to meet some requirements like up for x amount of time or having rules that have similar values of their relays. Also some of them require you to apply or reach out to the relay owner to request your server being added. 

Lastly you also want to make sure your adding the link that ends in /inbox or /actor and not just the domain you land on for the relay. 

Hopefully that helps out.
